### PR TITLE
Change auto deployment links

### LINKED ISSUE #2521: Use /deploy vs :4000 for autodeployment links
Recently, @AmasiaNalbandian [added a redirect in our nginx router](https://github.com/Seneca-CDOT/telescope/blob/c505e3ba6fc38407d91514d5906cf8f6278d3799/config/nginx.conf.template#L236-L239) from :4000 -> /deploy, such that you can do https://dev.telescope.cdot.systems:4000/status or https://dev.telescope.cdot.systems/deploy/status.  The latter is much nicer from a developer experience point of view.

In a few places, we are using :4000 and I think we should switch to /deploy:

- https://github.com/Seneca-CDOT/telescope/blob/17d5c1522e7b799d4d9b826ec0223c4bb3714ea9/src/api/status/public/js/build-log/api.js#L2
- https://github.com/Seneca-CDOT/telescope/blob/ec5ddd2f97118ed3d47536c91e91cac686002120/src/api/status/src/server.js#L22
- etc

NOTE: we don't need to change every use of :4000 to /deploy; only the ones where we are acting as a consumer of the deploy routes vs. defining them.

### PR DESCRIPTION
**Pull request by JerryHue**
## Issue This PR Addresses

Fixes #2521.

## Type of Change

- [X] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description

Update the consumer code for the status dashboard to present a more developer-friendly URL: from https://dev.telescope.cdot.systems:4000/status to https://dev.telescope.cdot.systems/deploy/status AND
https://dev.telescope.cdot.systems:4000/log to https://dev.telescope.cdot.systems/deploy/log.

## Checklist

- [X] **Quality**: This PR builds and passes our npm test and works locally.
- [ ] **Tests**: This PR does not require test as it does not change functionality.
- [ ] **Screenshots**: This PR does not provide screenshots as it does not change UI.
- [X] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed.

Co-Authored-By: Qiwen Yu <yvaineyuu.0201@gmail.com> (AKA @Qiwen-Yu)

### CODE REVIEW

File: tools/autodeployment/README.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ is created on GitHub).
 There are also a few other routes you can use to get information about the
 auto deployment server:
 
-- `GET :4000/status`: returns JSON data about the current status of the build.
+- `GET /deploy/status`: returns JSON data about the current status of the build.
 - `GET :4000/log`: returns a real-time stream of current build log output, if any.
```
Comment (humphd) [inline tools/autodeployment/README.md:18]: What about this one?

**Review by humphd — CHANGES_REQUESTED** [review]

File: tools/autodeployment/README.md (hunk 1 above)
Comment (JerryHue) [reply tools/autodeployment/README.md:18]: Oops! Silly mistake. I'm going to change it now.

**Review by DukeManh — DISMISSED** [review]

**Review by AmasiaNalbandian — DISMISSED** [review]
looks good!

**Comment by JerryHue** [comment]
I have rebased the branch. Let me know if we need anything else. @DukeManh, @AmasiaNalbandian, @humphd

**Review by humphd — DISMISSED** [review]

**Review by DukeManh — DISMISSED** [review]

**Comment by humphd** [comment]
@JerryHue land when ready!

**Review by Andrewnt219 — APPROVED** [review]

**Comment by JerryHue** [comment]
So, uh, because of another update to master, I had to rebase again... Should I force the squash and merge? I don't know if I have the permissions for it.

**Review by Andrewnt219 — APPROVED** [review]
"Squash and merge" is fine.

**Pull request merged**